test/typescript/index.ts

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+
+const equal = require("../..");
+
+type DeepObject = {
+  any: any;
+};
+
+type State = {
+  any: {
+    any: any;
+  };
+};
+
+// State
+const selector = (state: State) => {
+  a: {
+    b: [1, 2, 3];
+  }
+};
+
+class TestComponent extends React.Component {
+  constructor() {
+    super();
+    this.useSelector = this.useSelector;
+  }
+
+  shouldComponentUpdate(nextProps) {
+    return !equal(this.props, nextProps);
+  }
+
+  useSelector<TState, TSelected>(
```

Comment:
can we use the react-redux lib directly in our test so that we recreate the exact error? unless you think that would be too much overhead?